src/fairplay.js

```diff
@@ -45,21 +48,21 @@ const concatInitDataIdAndCertificate = ({initData, id, cert}) => {
 const addKey = ({video, contentId, initData, cert, options, getLicense, eventBus}) => {
   return new Promise((resolve, reject) => {
     if (!video.webkitKeys) {
-      video.webkitSetMediaKeys(new window.WebKitMediaKeys(FAIRPLAY_KEY_SYSTEM));
-    }
-
-    if (!video.webkitKeys) {
-      reject('Could not create MediaKeys');
-      return;
+      try {
+        video.webkitSetMediaKeys(new window.WebKitMediaKeys(FAIRPLAY_KEY_SYSTEM));
+      } catch (error) {
+        reject('Could not create MediaKeys');
+      }
     }
 
-    const keySession = video.webkitKeys.createSession(
-      'video/mp4',
-      concatInitDataIdAndCertificate({id: contentId, initData, cert}));
+    let keySession;
 
-    if (!keySession) {
-      reject('Could not create key session');
-      return;
+    try {
+      keySession = video.webkitKeys.createSession(
+        'video/mp4',
+        concatInitDataIdAndCertificate({id: contentId, initData, cert}));
+    } catch (error) {
+      reject('Could not create key session: invalid mimeType or initData');
```

Comment:
So while I was coding this, I was thinking that `videojs.xhr` would have an `error` if the `getCertificate` call returned some non-good `statusCode`, like a 500, in which case the `getCertificate` promise would be rejected. Of course, that's not the case, so the `getCertificate` promise would not be rejected, and the invalid data would be processed in `concatInitDataIdAndCertificate`.

What do you think is the best approach? I guess probably to add certificate errors to the list of possible problems here, since you can't know for sure that the certificate data returned by the server is valid.

Reply:
Fair enough.